src/modules/RA/DataLayer/components/tabs/StyleTab/index.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import {
+  FormTab,
+  ArrayInput,
+  SimpleFormIterator,
+  useTranslate,
+  useDataProvider,
+} from 'react-admin';
+
+import { useField } from 'react-final-form';
+import Typography from '@material-ui/core/Typography';
+
+import useSourceData from '../../useSourceData';
+import Placeholder from '../../../../../../components/Placeholder';
+import { RES_DATASOURCE } from '../../../../ra-modules';
+
+import CustomLayer from './CustomLayer';
+import StyleEditor from './StyleEditor';
+
+const StyleTab = ({ external, ...rest }) => {
+  const translate = useTranslate();
+
+  const dataProvider = useDataProvider();
+
+  const { geom_type: geomType, id: sourceId } = useSourceData('source');
+
+  const {
+    input: { value: fields },
+  } = useField('fields');
+
+  const { input: { value: mainStyle } } = useField('main_style');
+
+  const getValuesOfProperty = React.useCallback(property =>
+    dataProvider('PROPERTY_VALUES', RES_DATASOURCE, { id: sourceId, property }),
+  [dataProvider, sourceId]);
+
+  if (geomType === undefined || !sourceId) {
```

Comment:
Now we can use the nullish coalescing operator for this kind of condition
`if (geomType ?? !sourceId) {`

Reply:
Done